### PR TITLE
misc(core): fix tendermint config folder delete

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

This attempts to fix the CI flakiness due to the tendermint config folder delete.

This fix will be merged in the app with this PR https://github.com/celestiaorg/celestia-app/pull/1426 and then this change can be reverted.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: core/testing.go
Comment (renaynay) [inline core/testing.go:166]:
```suggestion
	return os.RemoveAll(rootDir)
```

**Review by renaynay — APPROVED** [review]

**Comment by renaynay** [comment]
Thank you @SweeXordious 🙏🏻

File: core/testing.go
Comment (Wondertan) [inline core/testing.go:91]: I think this should be part of the cleanupCoreNode func returned from `StartNode`, as it's clearly buggy. Think this way:
* You provide and API to setup something
* Additionally you provide API to clean up this something, as this something writes to disk
* Users calls this cleanup API, but it turns out to be not enough and users had to manually cleanup additional things.
* Do we want every user to stumble upon this themselves or do we want to do proper cleanup in one place without making users even noticing such issues?

**Review by Wondertan — CHANGES_REQUESTED** [review]
Thanks for this, but this should live in app not in the node

File: core/testing.go
Comment (SweeXordious) [reply core/testing.go:91]: It is on app side:
https://github.com/celestiaorg/celestia-app/blob/b56c1bb292d8f4e16c4e55d47032551d23e2cf8e/testutil/testnode/full_node.go#L218

Except that it's for the default network, which, I guess, we expect the users to use if they don't wanna do any extra step.
It's a good idea to add it to `StartNode`. However, I added it here since you seem to depend on releases and not sure when next Celestia-app release will, and your CI will stay flaky until you update.

I will open a PR for this:
> I think this should be part of the cleanupCoreNode func returned from StartNode

And if you prefer to wait for this to come from next release, we can close this one :D

Comment (SweeXordious) [reply core/testing.go:91]: PR on app side: https://github.com/celestiaorg/celestia-app/pull/1426

Comment (Wondertan) [reply core/testing.go:91]: Ok, it would have been nice to mention in the PR desc that this fix was merged into the app. The idea to have it temporarily here makes sense to me. Pls add the TODO here to remove once we update the app and we can merge this one

File: core/testing.go
Comment (SweeXordious) [inline core/testing.go:149]:
```suggestion
// and delete them one by one.
// TODO: remove after updating the app to a version that contains this PR https://github.com/celestiaorg/celestia-app/pull/1426
```

File: core/testing.go
Comment (SweeXordious) [reply core/testing.go:91]: [4faafcf](https://github.com/celestiaorg/celestia-node/pull/1822/commits/4faafcf19fcf837bc6945b2df1bb815f34b3b460)

**Review by Wondertan — APPROVED** [review]
Thanks @SweeXordious !

**Comment by Wondertan** [comment]
Also, do we have any confirmationyet  that this actually fixes the problem? @SweeXordious

**Comment by SweeXordious** [comment]
I tried it on celestia-app and orchestrator-relayer and I don't see the flakiness that often.

However, if the flakiness occurs, with this PR, we will know exactly which file is causing it and we will be able to further debug it. Because, currently, all we know is that the `config` folder is what's causing it. And this folder contains many files and is used by my routines. So we can't know for sure which resource is leaking or not closing properly.

I was hoping to see the flakiness on those repos to further debug. But it just disappeared :D which is a good thing and made me propose this change to you so it reduces, if not fix, the flakiness.

**Comment by SweeXordious** [comment]
This change will be included into v0.12 release: https://github.com/celestiaorg/celestia-app/pull/1429

**Comment by Wondertan** [comment]
Maybe we can just wait for it then, instead of emerging, as it will be very soon(in a few days)

**Comment by renaynay** [comment]
Closing in favour of #1843 which contains this change

**Pull request closed without merging**
